auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.kt

```diff
@@ -172,6 +172,39 @@ public class AuthenticationAPIClientTest {
         assertThat(body, Matchers.not(Matchers.hasKey("connection")))
     }
 
+    @Test
+    public fun shouldLoginWithMFARecoveryCode() {
+        mockAPI.willReturnSuccessfulLoginWithRecoveryCode()
+        val callback = MockAuthenticationCallback<Credentials>()
+        val auth0 = auth0
+        val client = AuthenticationAPIClient(auth0)
+        client.loginWithRecoveryCode("ey30.the-mfa-token.value", "123456")
+            .start(callback)
+        ShadowLooper.idleMainLooper()
+        assertThat(
+            callback, AuthenticationCallbackMatcher.hasPayloadOfType(
+                Credentials::class.java
+            )
+        )
+        assertThat(callback.payload.recoveryCode, Matchers.`is`("654321"))
```

Comment:
If we're asserting the recovery code on the payload, should we be asserting the token value as well?

Reply:
Unless that is asserted in some other test, but the same could be said about the `Accept-Language` assert.

Reply:
@Widcket This line checks for the content of the `response`. Here we only care about checking that if a new recovery code was sent, we can parse it. (see one is 123456 and the other is 654321).

A few lines below we assert the content of the `request`, which includes both mfa token and recovery code.